tests/unit/test_charm.py

```diff
@@ -126,343 +161,196 @@ def test_given_container_can_connect_and_fiveg_nrf_relation_is_not_created_when_
-        (root / "etc/udm/home_network.key").write_text(home_network_private_key)
-        (root / "support/TLS/udm.pem").write_text(STORED_CERTIFICATE)
         (root / f"etc/udm/{CONFIG_FILE_NAME}").write_text("super different config file content")
-
         self._create_certificates_relation()
-        patch_check_output.return_value = POD_IP
-        patched_nrf_url.return_value = VALID_NRF_URL
         nrf_relation_id = self._create_nrf_relation()
-
         self.harness.container_pebble_ready(self.container_name)
 
         self.harness.remove_relation(nrf_relation_id)
         self.harness.evaluate_status()
-        self.assertEqual(
-            self.harness.model.unit.status,
-            BlockedStatus("Waiting for fiveg_nrf relation"),
-        )
+
+        assert self.harness.model.unit.status == BlockedStatus("Waiting for fiveg_nrf relation")
 
     def test_given_container_can_connect_and_fiveg_nrf_relation_is_created_and_not_available_when_configure_sdcore_udm_then_status_is_waiting(  # noqa: E501
-        self,
+        self, add_storage
     ):
-        self.harness.add_storage(storage_name="certs", attach=True)
         self.harness.set_can_connect(container=self.container_name, val=True)
+        self.mock_nrf_url.return_value = None
         self._create_nrf_relation()
         self._create_certificates_relation()
 
         self.harness.charm._configure_sdcore_udm(event=Mock())
         self.harness.evaluate_status()
-        self.assertEqual(
-            self.harness.model.unit.status,
-            WaitingStatus("Waiting for NRF endpoint to be available"),
-        )
 
-    @patch("charms.sdcore_nrf_k8s.v0.fiveg_nrf.NRFRequires.nrf_url", new_callable=PropertyMock)
-    def test_given_config_storage_is_not_attached_when_configure_sdcore_udm_then_status_is_waiting(  # noqa: E501
-        self, patched_nrf_url
-    ):
-        self.harness.set_can_connect(container=self.container_name, val=True)
-        self.harness.add_storage(storage_name="certs", attach=True)
+        assert self.harness.model.unit.status == WaitingStatus(
+                                                    "Waiting for NRF endpoint to be available"
+                                                )
 
-        patched_nrf_url.return_value = VALID_NRF_URL
-        self._create_nrf_relation()
-        self._create_certificates_relation()
-
-        self.harness.charm._configure_sdcore_udm(event=Mock())
-        self.harness.evaluate_status()
-        self.assertEqual(
-            self.harness.model.unit.status, WaitingStatus("Waiting for the storage to be attached")
-        )
-
-    @patch("charms.sdcore_nrf_k8s.v0.fiveg_nrf.NRFRequires.nrf_url", new_callable=PropertyMock)
-    def test_given_certs_storage_is_not_attached_when_configure_sdcore_udm_then_status_is_waiting(  # noqa: E501
-        self, patched_nrf_url
+    @pytest.mark.parametrize(
+        "storage_name",
+        [
+            ("certs"),
+            ("config"),
+        ]
+    )
```

Comment:
```suggestion
    @pytest.mark.parametrize(
        "storage_name",
        [
            "certs",
            "config",
        ]
    )
```

Parenthesis are not required.